src/main/java/io/cdap/plugin/gcp/bigquery/sqlengine/BigQueryJoinDataset.java

```diff
@@ -72,7 +69,7 @@ private BigQueryJoinDataset(String datasetName,
     this.bqDataset = bqDataset;
     this.bqTable = bqTable;
     this.jobId = jobId;
-    this.helper = new BigQuerySQLBuilder(this.project, this.bqDataset, stageToTableNameMap);
+    this.helper = new BigQuerySQLBuilder(this.joinDefinition, this.project, this.bqDataset, stageToTableNameMap);
```

Comment:
please name this `queryBuilder` or similar. It's better to have specific names

Reply:
Good point, let me rename this.